File: NutriSmart/app/src/main/java/com/example/nutrismart/data/repository/WaterRepository.kt
package com.example.nutrismart.data.repository

import com.example.nutrismart.data.SessionManager
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class WaterRepository @Inject constructor(
    private val sessionManager: SessionManager
) {
    fun getWaterIntake(): Int = sessionManager.getWaterIntake()

    fun saveWaterIntake(water: Int) {
        sessionManager.saveWaterIntake(water)
    }
}

File: NutriSmartIOS/data/repository/WaterRepository.swift
//
//  WaterRepository.swift
//  NutriSmartIOS
//

import Foundation

class WaterRepository {
    static let shared = WaterRepository()
    
    private let sessionManager = SessionManager.shared
    
    private init() {}
    
    func getWaterIntake() -> Int { sessionManager.getWaterIntake() }
    
    func saveWaterIntake(water: Int) {
        sessionManager.saveWaterIntake(ml: water)
    }
}
